### PR TITLE
Change enableAccesslog in the startup parameter to enableMetric

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:
Change `enableAccesslog` in the startup parameter to `enableMetric`.
The ability to modify `enableMetric` will be added to kmeshctl in the future.

**Which issue(s) this PR fixes**:
A part of #1057 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: daemon/options/bpf.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func (c *BpfConfig) AttachFlags(cmd *cobra.Command) {
 	cmd.PersistentFlags().StringVar(&c.Mode, "mode", "dual-engine", "controller plane mode, valid values are [kernel-native, dual-engine]")
 	cmd.PersistentFlags().BoolVar(&c.EnableMda, "enable-mda", false, "enable mda")
 	cmd.PersistentFlags().BoolVar(&c.EnableBpfLog, "enable-bpf-log", false, "enable ebpf log in daemon process")
-	cmd.PersistentFlags().BoolVar(&c.EnableAccesslog, "enable-accesslog", false, "enable accesslog in daemon process")
+	cmd.PersistentFlags().BoolVar(&c.EnableMonitoring, "monitoring", false, "enable kmesh traffic monitoring in daemon process")
```
Comment (hzxuzhonghu) [inline daemon/options/bpf.go:44]: should we make it on by default?

If EnableMonitoring=false and EnableBpfLog = true, that's not valid, right?

We should validate them

File: bpf/kmesh/workload/sockops.c [hunk 2]
```diff
@@ -257,7 +269,9 @@ int sockops_prog(struct bpf_sock_ops *skops)
     case BPF_SOCK_OPS_ACTIVE_ESTABLISHED_CB:
         if (!is_managed_by_kmesh(skops))
             break;
-        observe_on_connect_established(skops->sk, OUTBOUND);
+        if (enable_metric()) {
+            observe_on_connect_established(skops->sk, OUTBOUND);
```
Comment (nlgwcy) [inline bpf/kmesh/workload/sockops.c:273]: `enable_metric` should be called in `observe_on_connect_established`

File: daemon/options/bpf.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ type BpfConfig struct {
 	Cgroup2Path     string
 	EnableMda       bool
 	EnableBpfLog    bool
-	EnableAccesslog bool
+	EnableMetric    bool
```
Comment (nlgwcy) [inline daemon/options/bpf.go]: Does `enableMetric` control both metric and access log? `EnableMetric` -> `EnableObserve`？

File: ctl/monitoring/monitoring.go [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+)
+
+var log = logger.NewLoggerScope("kmeshctl/monitoring")
+
+func NewCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "monitoring",
+		Short: "Control Kmesh's monitoring to be turned on as needed",
+		Example: `# Enable Kmesh's accesslog:
+kmeshctl monitoring <kmesh-daemon-pod> --accesslog enable
```
Comment (hzxuzhonghu) [inline ctl/monitoring/monitoring.go:45]: provide an example with no pod specified

File: daemon/options/bpf.go (hunk 1 above)
Comment (LiZhenCheng9527) [reply daemon/options/bpf.go]: `EnableMonitoring` and `EnableBpfLog` are not related.
`EnableMonitoring` control Kmesh monitoring. And I will add a function to control accesslog in `kmeshctl`.
```sh
# Enable Kmesh's accesslog:
kmeshctl monitoring <kmesh-daemon-pod> --accesslog enable

# Disable Kmesh's accesslog:
kmeshctl monitoring <kmesh-daemon-pod> --accesslog disable

# Enable Kmesh's metrics and accesslog:
kmeshctl monitoring <kmesh-daemon-pod> --all enable

# Disable Kmesh's metrics and accesslog:
kmeshctl monitoring <kmesh-daemon-pod> --all disable
```

File: daemon/options/bpf.go (hunk 3 above)
Comment (LiZhenCheng9527) [reply daemon/options/bpf.go]: Done

File: ctl/monitoring/monitoring.go (hunk 4 above)
Comment (LiZhenCheng9527) [reply ctl/monitoring/monitoring.go]: Done

File: bpf/kmesh/workload/sockops.c (hunk 2 above)
Comment (LiZhenCheng9527) [reply bpf/kmesh/workload/sockops.c:273]: Can you tell me why should do it?

File: ctl/monitoring/monitoring.go [hunk 5]
```diff
@@ -41,17 +41,15 @@ func NewCmd() *cobra.Command {
...
+kmeshctl monitoring <kmesh-daemon-pod> --all enable/disable
 
-# Enable Kmesh's metrics and accesslog:
-kmeshctl monitoring <kmesh-daemon-pod> --all enable
-
-# Disable Kmesh's metrics and accesslog:
-kmeshctl monitoring <kmesh-daemon-pod> --all disable`,
+# If you want to change the monitoring functionality of all kmesh daemons in the cluster
+kmeshctl monitoring --accesslog enable/disable
+kmeshcrl monitoring --all enable/disable`,
```
Comment (hzxuzhonghu) [inline ctl/monitoring/monitoring.go:52]:
```suggestion
kmeshctl monitoring --all enable/disable`,
```

File: bpf/kmesh/workload/sockops.c (hunk 2 above)
Comment (LiZhenCheng9527) [reply bpf/kmesh/workload/sockops.c]: Done

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**